src/main/java/valoeghese/biomeoverhaul/world/feature/BorealFeature.java

```diff
@@ -18,109 +19,109 @@
 {
 	private static final BlockState LOG, LEAVES;
 
-	public BorealFeature(Function<Dynamic<?>, ? extends DefaultFeatureConfig> function_1)
+	public BorealFeature(Function<Dynamic<?>, ? extends DefaultFeatureConfig> config)
 	{
-		super(function_1, false, 12, 4, LOG, LEAVES);
+		super(config, false, 12, 4, LOG, LEAVES);
 	}
 
-	public boolean generate(Set<BlockPos> set_1, ModifiableTestableWorld modifiableTestableWorld_1, Random random_1, BlockPos blockPos_1)
+	public boolean generate(Set<BlockPos> positions, ModifiableTestableWorld world, Random rand, BlockPos pos, MutableIntBoundingBox bb)
 	{
-		int int_1 = this.getHeight(random_1);
-		if (!this.checkTreeFitsAndReplaceGround(modifiableTestableWorld_1, blockPos_1, int_1))
+		int height = this.getHeight(rand);
+		if (!this.checkTreeFitsAndReplaceGround(world, pos, height))
 		{
 			return false;
 		} else
 		{
-			this.makeTopLeaves(modifiableTestableWorld_1, blockPos_1.getX(), blockPos_1.getZ(), blockPos_1.getY() + int_1, 0, random_1);
+			this.makeTopLeaves(world, pos.getX(), pos.getZ(), pos.getY() + height, 0, rand, bb, positions);
 
-			for(int int_2 = 0; int_2 < int_1; ++int_2)
+			for(int i = 0; i < height; ++i)
 			{
-				if (isAirOrLeaves(modifiableTestableWorld_1, blockPos_1.up(int_2)))
+				if (isAirOrLeaves(world, pos.up(i)))
 				{
-					this.setBlockState(set_1, modifiableTestableWorld_1, blockPos_1.up(int_2), this.log);
+					this.setBlockState(positions, world, pos.up(i), this.log, bb);
 				}
 
-				if (int_2 < int_1 - 1)
+				if (i < height - 1)
 				{
-					if (isAirOrLeaves(modifiableTestableWorld_1, blockPos_1.add(1, int_2, 0)))
+					if (isAirOrLeaves(world, pos.add(1, i, 0)))
 					{
-						this.setBlockState(set_1, modifiableTestableWorld_1, blockPos_1.add(1, int_2, 0), this.log);
+						this.setBlockState(positions, world, pos.add(1, i, 0), this.log, bb);
 					}
 
-					if (isAirOrLeaves(modifiableTestableWorld_1, blockPos_1.add(1, int_2, 1)))
+					if (isAirOrLeaves(world, pos.add(1, i, 1)))
 					{
-						this.setBlockState(set_1, modifiableTestableWorld_1, blockPos_1.add(1, int_2, 1), this.log);
+						this.setBlockState(positions, world, pos.add(1, i, 1), this.log, bb);
 					}
 
-					if (isAirOrLeaves(modifiableTestableWorld_1, blockPos_1.add(0, int_2, 1)))
+					if (isAirOrLeaves(world, pos.add(0, i, 1)))
 					{
-						this.setBlockState(set_1, modifiableTestableWorld_1, blockPos_1.add(0, int_2, 1), this.log);
+						this.setBlockState(positions, world, pos.add(0, i, 1), this.log, bb);
 					}
 				}
 			}
 
-			this.replaceGround(modifiableTestableWorld_1, random_1, blockPos_1);
```

Comment:
also I'm probably going to copy parts of the code I need then close this instead of merge, since I've refactored a whole bunch of stuff internally

Reply:
Sounds good!